framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -1128,6 +1155,34 @@ public AnnotatedTypeMirror getType(AnnotatedTypeMirror type, AnnotatedTypeFactor
 
       return this.type;
     }
+    /**
+     * Adds a declaration annotation to this field declaration and returns whether it was a new
+     * annotation.
+     *
+     * @param annotation declaration annotation to add
+     * @return true if {@code annotation} wasn't previously stored for this field declaration
+     */
+    public boolean addDeclarationAnnotation(AnnotationMirror annotation) {
+      if (declarationAnnotations == null) {
+        declarationAnnotations = new HashSet<>();
+      }
+
+      return declarationAnnotations.add(annotation);
+    }
+
+    /**
+     * Returns the inferred declaration annotations on this field, or null if there are no
+     * annotations.
```

Comment:
This method seems to return the empty set if there are no declaration annotations, not null. I think that the code is right and this specification is wrong, but you should check to be sure, and then make the appropriate fix.

Reply:
I updated this to be "an empty set"